src/snmalloc/pal/pal_linux.h

```diff
@@ -8,6 +8,11 @@
 #  include <sys/mman.h>
 #  include <sys/prctl.h>
 #  include <syscall.h>
+// __has_include does not reliably determine if we actually have linux/random.h
+// available
+#  if defined(SNMALLOC_HAS_LINUX_RANDOM_H)
```

Comment:
What's going on with `__has_include`?

It might also be worth a comment here to point out that below we check for `SYS_getrandom`, which won't be defined if this file isn't included?

Reply:
I honestly cannot tell you what is going on with `__has_include` other than with gcc 9.3 is wasn't picking up the following:

```c
#if __has_include(<linux/random.h>)
#   include <linux/random.h>
#endif
```

I honestly have no idea why.

Reply:
`SYS_getrandom` seems to come from `<syscall.h>`

Reply:
But overall if both `sys/random.h` and `linux/random.h` aren't available, the build will fail.